src/reducers/filterReducer.js

```diff
@@ -1,15 +1,83 @@
 import * as filterActions from 'actions/filterActions'
+import { combineReducers } from 'redux'
 
-export const filter = (defaultFilter) => (state = defaultFilter, action) => {
+export const incidentId = (defaultIncidentId) => (state = defaultIncidentId, action) => {
   switch (action.type) {
     case filterActions.CLEAR_EVENT_FILTER_INCIDENTID:
-      const {incidentId, ...newFilter} = action.filter
-      return newFilter
+      return null
     case filterActions.CHANGE_EVENT_FILTER:
-      return {...state, eventTypes: [], ticketId: null, ...action.filter}
+      if (action.filter.incidentId) {
+        return action.filter.incidentId
+      } else return state
     default:
       return state
   }
 }
 
+export const eventTypes = (defaultEventTypes) => (state = defaultEventTypes, action) => {
+  switch (action.type) {
+    case filterActions.CHANGE_EVENT_FILTER:
+      if (action.filter.eventTypes) {
+        return action.filter.eventTypes
+      } else return state
+    default:
+      return state
+  }
+}
+
+export const startTime = (defaultStartTime) => (state = defaultStartTime, action) => {
+  switch (action.type) {
+    case filterActions.CHANGE_EVENT_FILTER:
+      if (action.filter.startTime) {
+        return action.filter.startTime
+      } else return state
+    default:
+      return state
+  }
+}
+
+export const endTime = (defaultEndTime) => (state = defaultEndTime, action) => {
+  switch (action.type) {
+    case filterActions.CHANGE_EVENT_FILTER:
+      if (action.filter.endTime) {
+        return action.filter.endTime
+      } else return state
+    default:
+      return state
+  }
+}
+
+export const dataSearch = (defaultDataSearch) => (state = defaultDataSearch, action) => {
+  switch (action.type) {
+    case filterActions.CHANGE_EVENT_FILTER:
+      if (action.filter.DataSearch) {
+        return action.filter.DataSearch
+      } else return state
+    default:
+      return state
+  }
+}
+
+export const ticketId = (defaultTicketId) => (state = defaultTicketId, action) => {
+  switch (action.type) {
+    case filterActions.CHANGE_EVENT_FILTER:
+      if (action.filter.ticketId) {
+        return action.filter.ticketId
+      } else return state
+    default:
+      return state
+  }
+}
+
+export const filter = (defaultFilter) => {
```

Comment:
Instead of checking if `defaultFilter` is null or undefined in each reducer to combine, you could set a default at the function signature, something like this:
```
const initialDefaultFilter = { something: something }
export const filter = (defaultFilter = initialDefaultFilter) => {
```

Then you could just do this in each reducer to combine:
```
ticketId: ticketId(defaultFilter.ticketId),
```

Reply:
export const filter = (defaultFilter = {}) => {

should be good enough

Reply:
this one actually doesn't work. For some reason, the defaultFilter is not undefined, there is a null value passed into it. So the default parameter will not be assigned to defaultFilter.

Reply:
who is calling that method, and what value is it passing into it?

Reply:
IIRC, that would be the FilterService method that extracts a filter object from the URL, which is called by configureStore.js. In reducers/index.js,

```JavaScript
  events: events(filters),
```
in the combineReducers statement can become

```JavaScript
  events: filters ? events(filters) : events(),
```

To get around this, though there are other ways to get the same effect. Not sure what's best.

Reply:
so I introduced the initalDefaultFilter, but it broke the filterReducerTest because in the test it's testing filterReducer as a whole. So, should I just rewrite the test to test each reducer in filterReducer?